core/templates/pages/exploration-player-page/current-lesson-player/learner-experience/tutor-card.component.spec.ts

```diff
@@ -273,9 +287,186 @@ describe('Tutor card component', () => {
     expect(deviceInfoService.isMobileDevice).toHaveBeenCalled();
     expect(audioBarStatusService.isAudioBarExpanded).toHaveBeenCalled();
     expect(urlInterpolationService.getStaticImageUrl).toHaveBeenCalled();
+    expect(componentInstance.completedChaptersCount).toBe(1);
+    expect(componentInstance.shouldShowProgressBar).toBe(false);
     expect(componentInstance.getInputResponsePairId).toHaveBeenCalled();
   }));
 
+  it('should set the chapterCount = 0 and throw error when user is unauthorized', fakeAsync(() => {
+    const error401 = {status: 401, message: 'Unauthorized'};
+    spyOn(
+      chapterProgressService,
+      'updateCompletedChaptersCount'
+    ).and.returnValue(Promise.reject(error401));
+    spyOn(console, 'error').and.stub();
+
+    componentInstance.ngOnInit();
+    tick();
+
+    expect(componentInstance.completedChaptersCount).toBe(0);
+    expect(console.error).toHaveBeenCalledWith(
+      'User is logged out setting default chapters count'
+    );
+  }));
+
+  it('should set chapterCount = 0 and log error for other errors', fakeAsync(() => {
+    const error500 = {status: 500, message: 'Internal Server Error'};
+
+    componentInstance.displayedCard = mockDisplayedCard;
+
+    spyOn(
+      chapterProgressService,
+      'updateCompletedChaptersCount'
+    ).and.returnValue(Promise.reject(error500));
+    spyOn(console, 'error').and.stub();
+
+    spyOnProperty(
+      conversationFlowService,
+      'onOppiaFeedbackAvailable',
+      'get'
+    ).and.returnValue(new Subject<void>());
+
+    spyOn(
+      componentInstance,
+      'setNextMilestoneAndCheckIfProgressBarIsShown'
+    ).and.returnValue(false);
```

Comment:
```suggestion
expect(componentInstance.shouldShowProgressBar).toBe(false);
```
Since we are no longer using calling this method.

Reply:
Hi for this I will make the changes 👍.